File: Curso_C#/Proyecto_Backend/MindFit_Intelligence_Backend/MindFit_Intelligence_Backend/Models/Ejercicio.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MindFit_Intelligence_Backend.Models
{
    public class Ejercicio
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int IdEjercicio { get; set; }

        [Column(TypeName = "VARCHAR(100)")]
        public required string DescEjercicio { get; set; }

        [ForeignKey(nameof(GrupoMuscular))]
        public int IdGrupoMuscular { get; set; }
        public GrupoMuscular GrupoMuscular { get; set; } = default!;

        [ForeignKey(nameof(TipoEjercicio))]
        public int IdTipoEjercicio { get; set; }
        public TipoEjercicio TipoEjercicio { get; set; } = default!;

        [ForeignKey(nameof(Maquina))]
        public int? IdMaquina { get; set; }
        public Maquina? Maquina { get; set; }

        [ForeignKey(nameof(Equipamiento))]
        public int? IdEquipamiento { get; set; }
        public Equipamiento? Equipamiento { get; set; }
    }
}

File: Curso_CSharp/Otras Versiones/V9_Proyecto_MindFit_Intelligence_PreModuloIA/Proyecto_MindFit_Intelligence/Frontend/src/types/ejercicio.ts
import type { EquipamientoDto } from './equipamiento'
import type { MaquinaDto } from './maquina'

export type Musculo =
  | 'Pecho'
  | 'Espalda'
  | 'Cuadriceps'
  | 'Biceps'
  | 'Triceps'
  | 'Gluteos'
  | 'Abdomen'
  | 'Hombros'
  | 'Gemelos'
  | 'Antebrazos'
  | 'Lumbares'
  | 'Isquiotibiales'

export type TipoDeEjercicio = 'Calentamiento' | 'Entrenamiento' | 'Estiramiento'

export interface GrupoMuscularDto {
  idGrupoMuscular: number
  nombreMusculo: Musculo
  idMapaAnatomico: string | null
}

export interface TipoEjercicioDto {
  idTipoEjercicio: number
  nombreTipo: TipoDeEjercicio
}

export interface EjercicioDto {
  idEjercicio: number
  descEjercicio: string
  grupoMuscular: GrupoMuscularDto
  tipoEjercicio: TipoEjercicioDto
  maquina: MaquinaDto | null
  equipamiento: EquipamientoDto | null
}

export interface EjercicioInsertDto {
  descEjercicio: string
  idGrupoMuscular: number
  idTipoEjercicio: number
  idMaquina: number | null
  idEquipamiento: number | null
}

export interface EjercicioUpdateDto {
  descEjercicio: string
  idGrupoMuscular: number
  idTipoEjercicio: number
  idMaquina: number | null
  idEquipamiento: number | null
}
